eZ/Publish/Core/FieldType/Date/SearchField.php

```diff
@@ -83,4 +83,40 @@ public function getDefaultSortField()
     {
         return $this->getDefaultMatchField();
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getFullTextData(Field $field, FieldDefinition $fieldDefinition)
+    {
+        // The field type stores date value as a timestamp of the start of the day in the
+        // environment's timezone.
+        // We format this as Y-m-d and add Z to signify UTC (zero offset).
+        $dateTime = new DateTime();
+        $dateTime->setTimestamp($field->value->data['timestamp']);
+
+        return [
+            $dateTime->format('Y-m-d\\Z'),
```

Comment:
This was not done originally, with changes like this we should rather not add in new stuff as well.
*(BTW this also means it could be integration tested, we already have a framework for that in place)*